whelktool/scripts/syslib/litteraturbanken-create-links-holdings.groovy

```diff
@@ -0,0 +1,201 @@
+PrintWriter failedHoldIDs = getReportWriter("Failed-to-create-holds.txt")
+PrintWriter failedModified = getReportWriter("Failed-to-modify-bibs.txt")
+
+File input = new File(scriptDir, "littbanken_hold_link.csv")
+List<String> ProgramLines = input.readLines().drop(1)
+
+def itemList = []
+
+for (String operation : ProgramLines) {
+    
+    String[] part = operation.split(';', -1)
+    String librisCN = part[0].trim()
+    String librisID = part[1].trim()
+    String status = part[2].trim() // not used
+    String url = part[3].trim()
+    String epub = part[4].trim()
+    String omslagsbild = part[5].replace("(större)","").trim()
+
+    String shortUrl = url.replaceFirst("https://","").trim()
+    String shortEpub = epub.replaceFirst("https://","").trim()
+	String normUrl = shortUrl.replaceAll("å","a").replaceAll("ä","a").replaceAll("ö","o")
+	String normEpub = shortEpub.replaceAll("å","a").replaceAll("ä","a").replaceAll("ö","o")
+    String shortCover = omslagsbild.replaceFirst("https://","").trim()
+
+    if (librisCN) {
+
+    where = """id in
+        (select lb.id
+        from lddb lb
+        where lb.collection = 'bib' and
+        lb.data#>>'{@graph,0,controlNumber}' = '${librisCN}'
+        and data#>>'{@graph,1,@type}' IN ('Electronic')
+    )"""
+    }
+    else {
+    where = """id = '${librisID}'"""
+    }
+
+    selectBySqlWhere(where, { bib ->
+        def bibMainEntity = bib.graph[1]["@id"]
+        def instance = bib.graph[1]
+        boolean foundUsage = false
+        boolean foundUrl = false
+        boolean foundEpub = false
+        boolean foundCover = false
+        
+        if (!instance["usageAndAccessPolicy"]) { // lägg till usageAndAccessPolicy om det inte finns
+            instance["usageAndAccessPolicy"] = []
+        }
+        
+        if (instance["usageAndAccessPolicy"]) {
+
+            instance["usageAndAccessPolicy"].each { it ->
+                asList(it["uri"]).each { uri -> 
+                    if (uri instanceof String && uri.contains("litteraturbanken.se/om/rattigheter")) { // kolla om uri matchar Litteraturbankens url om rättigheter
+                    foundUsage = true
+                	}	
+                }
+            }
+        }
+        
+        if (!instance["associatedMedia"]) { // lägg till associatedMedia om det inte finns
+            instance["associatedMedia"] = []
+        }
+        
+        if (instance["associatedMedia"]) {
+
+            instance["associatedMedia"].each { it ->
+                asList(it["uri"]).each { uri ->
+                    if (uri instanceof String && uri.contains("${shortUrl}")) { // kolla om uri matchar url från infilen (https:// borttaget då det kan vara sparat med http i Libris och vi vill inte skapa dubbletter)
```

Comment:
This works but there is no need for the quotes and brackets, i.e. `uri.contains(shortUrl)` would be just fine (since `shortUrl` is already a String)